File: Food.WebApi/Models/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Food.WebApi.Models
{
    public class Order
    {
        public Order() { }

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int Id { get; set; }
        public DateTime Date { get; set; }
        public decimal Amount { get; set; }
        public int Userid { get; set; }
        public List<int>? Count { get; set; }
        public ICollection<Dish>? Dishes { get; set; } = new List<Dish>();
    }
}

File: FoodToOrderAngular/src/app/models/order.ts
import { Cart } from "./cart"
import { Dish } from "./dish"

export class Order{
    id:number
    date:string
    amount:number
    userid:number
    dishes:Dish[]
    count:number[]

    constructor(id:number,date:string,a:number,u:number,d:Dish[],c:number[]){
        this.id=id
        this.date=date
        this.amount=a
        this.userid=u
        this.dishes=d
        this.count=c
    }
}